subprojects/execution/src/test/groovy/org/gradle/internal/execution/impl/steps/CacheStepTest.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.impl.steps
+
+import com.google.common.collect.ImmutableSortedMap
+import org.gradle.caching.internal.command.BuildCacheCommandFactory
+import org.gradle.caching.internal.controller.BuildCacheController
+import org.gradle.caching.internal.origin.OriginMetadata
+import org.gradle.internal.execution.CacheHandler
+import org.gradle.internal.execution.ExecutionOutcome
+import org.gradle.internal.execution.OutputChangeListener
+import org.gradle.internal.execution.UnitOfWork
+import org.gradle.internal.fingerprint.impl.EmptyCurrentFileCollectionFingerprint
+import org.gradle.internal.id.UniqueId
+import spock.lang.Specification
+
+class CacheStepTest extends Specification {
+
+    def buildCacheController = Mock(BuildCacheController)
+    def buildCacheCommandFactory = Mock(BuildCacheCommandFactory)
+    def outputChangeListener = Mock(OutputChangeListener)
+    def currentBuildId = UniqueId.generate()
+    def cacheStep = new CacheStep<CachingContext>(buildCacheController, outputChangeListener, buildCacheCommandFactory,
+        new SnapshotOutputStep<CachingContext>(currentBuildId,
+            new CatchExceptionStep<Context>(
+                new ExecuteStep(outputChangeListener)
+            )
+        )
+    )
```

Comment:
I don't think it's worth mocking halfway: why not delegate to a mock step as well?

Reply:
Delegating to a mock.